mkosi/config.py

```diff
@@ -1652,6 +1653,19 @@ class SettingScope(StrEnum):
     main = enum.auto()
     # Only passed down to tools tree, can only be configured in main image.
     tools = enum.auto()
+    # Only passed down to initrd, can only be configured in main image.
+    initrd = enum.auto()
+
+    def is_main_setting(self) -> bool:
+        return self in (SettingScope.main, SettingScope.tools, SettingScope.initrd, SettingScope.multiversal)
+
+    def removeprefix(self, setting: str) -> str:
+        if self == SettingScope.tools:
+            return setting.removeprefix("tools_tree_")
+        elif self == SettingScope.initrd:
+            return setting.removeprefix("initrd_")
+        else:
+            return setting
```

Comment:
Nit: now that it's inside a function
```suggestion

        return setting
```
would look a bit nicer.